src/components/Credential/Credential.tsx

```diff
@@ -37,22 +46,31 @@ export function CredentialVerifier({
     (async () => {
       if (!did || error) return;
 
-      if (Credential.isICredential(credential)) {
-        setClaimContents(credential.request.claim.contents);
-        isValid.set(await validateCredential(credential));
-        const attesterDid = credential.attestation.owner;
-        setAttester(await getW3NOrDid(attesterDid));
-        return;
+      if (isLegacyCredential(credential)) {
+        const realCredential = credential.request;
+        if (Credential.isICredential(realCredential)) {
+          setClaimContents(realCredential.claim.contents);
+          try {
+            await Credential.verifyCredential(realCredential);
```

Comment:
are you adding credentials or presentations to a signature? I would have expected the latter

Reply:
No, the signature is only for the hash of the files